### ISSUE TITLE
Implement Morpheme-Aware Training Script

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#44]
## Parent Issue

Part of #34 (Morpheme-Aware Training)

## Objective

Implement training script for morpheme-aware embeddings with three training objectives.

## Tasks

- [ ] Create `scripts/train_morpheme_embeddings_m1.py`
- [ ] Implement MorphemeDataset class
- [ ] Implement three training objectives:
  - [ ] Masked root prediction
  - [ ] Affix composition
  - [ ] Grammatical feature prediction
- [ ] Implement training loop with early stopping
- [ ] Add logging and checkpointing
- [ ] Add evaluation metrics

### 1. Masked Root Prediction

Learn root semantics from context.

```python
class MaskedRootPrediction(nn.Module):
    """Predict masked root from context + affixes."""
    
    def forward(self, context_words, affixes, true_root):
        # Encode context
        context_emb = self.encode_context(context_words)
        
        # Encode known affixes
        affix_emb = self.encode_affixes(affixes)
        
        # Predict root
        logits = self.predict_root(context_emb, affix_emb)
        
        # Loss
        return cross_entropy(logits, true_root)

# Example
# Input: "La [MASK]o estas granda kaj bela"
# Context: ["La", "estas", "granda", "kaj", "bela"]
# Affixes: {prefix: None, suffixes: [], ending: 'o'}
# True root: "dom" (house fits "large and beautiful")
```

### 2. Affix Composition

Learn how affixes transform root meanings.

```python
class AffixComposition(nn.Module):
    """Learn affix transformation semantics."""
    
    def forward(self, root, affixes, context):
        # Compose: root + affixes
        composed_emb = self.compose(root, affixes)
        
        # Should fit context
        context_emb = self.encode_context(context)
        
        # Contrastive loss
        similarity = cosine_similarity(composed_emb, context_emb)
        return contrastive_loss(similarity, positive=True)

# Example
# Root: "hund" (dog)
# Suffix: "ej" (place-for)
# Context: "La ___ estas granda" (fits kennel, not dog)
# Learns: -ej creates location semantics
```

### 3. Grammatical Feature Prediction

Learn grammatical patterns (not semantics).

```python
class GrammaticalFeaturePrediction(nn.Module):
    """Predict grammatical features from context."""
    
    def forward(self, context, true_features):
        # Encode context
        context_emb = self.encode_context(context)
        
        # Predict features
        case_logits = self.predict_case(context_emb)
        number_logits = self.predict_number(context_emb)
        tense_logits = self.predict_tense(context_emb)
        
        # Loss
        loss = (
            cross_entropy(case_logits, true_features['case']) +
            cross_entropy(number_logits, true_features['number']) +
            cross_entropy(tense_logits, true_features['tense'])
        )
        return loss

# Example
# Context: "Mi vidas la hund___"
# After transitive verb → predict accusative case
# Learns: Grammatical patterns, not semantics
```

## Combined Training Loop

```python
def train_epoch(model, dataloader, optimizer):
    for batch in dataloader:
        # Three objectives
        loss1 = masked_root_loss(batch)
        loss2 = composition_loss(batch)
        loss3 = grammatical_loss(batch)
        
        # Combined loss (equal weights)
        total_loss = loss1 + loss2 + loss3
        
        # Backward pass
        optimizer.zero_grad()
        total_loss.backward()
        optimizer.step()
        
    return total_loss
```

## Dataset

```python
class MorphemeDataset(Dataset):
    """Dataset for morpheme-aware training."""
    
    def __init__(self, corpus_path, affix_config):
        self.sentences = []
        self.affix_config = load_affix_config(affix_config)
        
        # Load corpus
        for entry in read_corpus(corpus_path):
            self.sentences.append(entry)
    
    def __getitem__(self, idx):
        sentence = self.sentences[idx]
        
        # For masked root prediction: mask random word
        masked_idx = random.randint(0, len(sentence['words']) - 1)
        masked_word = sentence['words'][masked_idx]
        
        # Extract morphemes
        morphemes = self.extract_morphemes(masked_word)
        
        # Classify affixes
        semantic_affixes = self.classify_affixes(
            morphemes, 
            self.affix_config
        )
        
        return {
            'context': sentence['words'],
            'masked_idx': masked_idx,
            'true_root': morphemes['root'],
            'affixes': semantic_affixes,
            'grammatical_features': morphemes['grammar']
        }
```

## Training Configuration

```bash
python scripts/train_morpheme_embeddings_m1.py \
  --corpus data/corpus_enhanced_m1.jsonl \
  --affix-config config/affix_classification.json \
  --root-vocab /tmp/root_frequency_analysis.json \
  --output models/morpheme_embeddings_m1 \
  --epochs 3 \
  --batch-size 64 \
  --lr 1e-4 \
  --max-samples 50000
```

**Expected duration**: 1-1.5 hours

## Success Criteria

- [x] Training script runs successfully
- [x] Three objectives implemented correctly
- [x] Loss decreases over epochs
- [x] Checkpoints saved regularly
- [x] Training completes in 1-1.5 hours
- [x] Final model saved to `models/morpheme_embeddings_m1/best_model.pt`

## Evaluation Metrics

Track during training:
- Masked root prediction accuracy (% correct predictions)
- Composition similarity (average cosine similarity for composed forms)
- Grammatical accuracy (% correct case/tense predictions)

## Time Estimate

2-3 hours implementation + 1-1.5 hours training

## Dependencies

- Requires: Issue #41 (root frequency analysis)
- Requires: Issue #42 (affix classification)
- Requires: Issue #43 (MorphemeAwareEmbedding model)
- Blocks: Issue #45 (evaluation)

**Comment by marctjones** [comment]
Closing: Morpheme-aware approach superseded by compositional embeddings in Stage 1. See #100 (Training Plan v3).
